fakeservice/README.md

```diff
@@ -0,0 +1,184 @@
+This object ([`ComputeAPI`](https://github.com/openstack/nova/blob/51ed40a6a5fc046cef35337980a1fc5ad704a421/nova/compute/rpcapi.py#L73)) define
+a [`router` method](https://github.com/openstack/nova/blob/51ed40a6a5fc046cef35337980a1fc5ad704a421/nova/compute/rpcapi.py#L383) who will return
+a [rpc client](https://github.com/openstack/nova/blob/51ed40a6a5fc046cef35337980a1fc5ad704a421/nova/compute/rpcapi.py#L405).
+
+The returned [RPC client](https://github.com/openstack/nova/blob/244c9240671d98b0df25b0ad0795b5de0c0c422c/nova/rpc.py#L208) returned by the [nova rpc module](https://github.com/openstack/nova/blob/master/nova/rpc.py) is an [oslo.messaging rpc client](https://github.com/openstack/nova/blob/master/nova/rpc.py#L18)
+
+The instantiated oslo.messaging [`RPCClient`](https://github.com/openstack/oslo.messaging/blob/master/oslo_messaging/rpc/client.py#L239) 
+
+The transport layer is defined by [nova](https://github.com/openstack/nova/blob/244c9240671d98b0df25b0ad0795b5de0c0c422c/nova/rpc.py#L69) and [it will be retrieved](https://github.com/openstack/nova/blob/244c9240671d98b0df25b0ad0795b5de0c0c422c/nova/rpc.py#L259) by using [the oslo.messaging mechanismes based on the config and the url](https://github.com/openstack/oslo.messaging/blob/master/oslo_messaging/transport.py#L218) and the oslo.messaging [defined drivers](https://github.com/openstack/oslo.messaging/blob/master/setup.cfg#L38,L50) by using [stevedore](https://github.com/openstack/oslo.messaging/blob/master/oslo_messaging/transport.py#L204).
+
+In our case we will instantiate a [`RabbitDriver`](https://github.com/openstack/oslo.messaging/blob/master/oslo_messaging/_drivers/impl_rabbit.py#L1266).
+
+The used driver will initiate a [connection pool](https://github.com/openstack/oslo.messaging/blob/master/oslo_messaging/_drivers/impl_rabbit.py#L1299,L1301)
+by using the [Connection class](https://github.com/openstack/oslo.messaging/blob/master/oslo_messaging/_drivers/impl_rabbit.py#L417) defined in the driver.
+
+The oslo.messaging connection pool module [will create the connection](https://github.com/openstack/oslo.messaging/blob/master/oslo_messaging/_drivers/pool.py#L144)
+and also start the healt check mechanism by [triggering the heartbeat in a dedicated thread](Also://github.com/openstack/oslo.messaging/blob/master/oslo_messaging/_drivers/impl_rabbit.py#L897)
+
+The nova rpc module also define a [RPC server](https://github.com/openstack/nova/blob/244c9240671d98b0df25b0ad0795b5de0c0c422c/nova/rpc.py#L215) inherited from [oslo.messaging](https://github.com/openstack/nova/blob/244c9240671d98b0df25b0ad0795b5de0c0c422c/nova/rpc.py#L18)
+
+There the used executor is an [`eventlet` executor](https://github.com/openstack/nova/blob/244c9240671d98b0df25b0ad0795b5de0c0c422c/nova/rpc.py#L226), so the initiated object will use eventlet and so the heartbeat thread will use a gthread.
```

Comment:
Just to be clear, "gthread" == "greenthread", correct?

I point this out mostly because gevent is a related project to this topic (that we don't use) and "gthread" sounds like it might be a gevent thing (it's not, from a quick glance at their docs).